### PR TITLE
DOMA-2726 disappearing of number on register layout

### PR DESCRIPTION
**Pull request by gegius**
https://doma.atlassian.net/browse/DOMA-2726

### CODE REVIEW

**Comment by gegius** [comment]
https://doma.atlassian.net/browse/DOMA-2726

https://doma.atlassian.net/browse/DOMA-2724 (?)

The problem is very old. Register as change-password use a validatePhoneForm. Apparently, in the RegisterContextProvider, when the captcha fails, we erase the user's phone number (why?)
We erase it not even at the stage of working with sending a message, but just like that

As we say, the author quit a long time ago of his own free will

File: apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx [hunk 1]
```diff
@@ -143,13 +143,14 @@ export const ValidatePhoneForm = ({ onFinish, onReset, title }): React.ReactElem
...
-    useEffect(() => {
-        if (isPhoneVisible) {
-            setShowPhone(formatPhone(phone))
-        } else {
-            const unHidden = formatPhone(phone)
-            setShowPhone(`${unHidden.substring(0, 9)}***${unHidden.substring(12)}`)
-        }
+    const handleNumberVisible = useCallback(() => {
+        const formattedPhone = formatPhone(phone)
+        isPhoneVisible
```
Comment (akarjakin) [inline apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx:148]: please use ternary operator only for providing values, in this case you need to use classical `if`

Comment (akarjakin) [reply apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx:148]: thou you're calling the same functions, other possibility here is to precalculate value using ternary operator and then call the function once.

Comment (akarjakin) [reply apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx:148]:
```
const phoneValue = isPhoneVisible ? formattedPhone : `${formattedPhone.substring(0, 9)}***${formattedPhone.substring(12)}`

setShowPhone(phoneValue)
```

File: apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx [hunk 2]
```diff
@@ -177,12 +178,11 @@ export const ValidatePhoneForm = ({ onFinish, onReset, title }): React.ReactElem
...
                                         phone: (
                                             <span style={{ whiteSpace: 'nowrap' }}>
-                                                {`${showPhone} `}
+                                                {`${formatPhone(showPhone)} `}
                                                 <Typography.Link
                                                     underline
                                                     style={{ color: 'black' }}
-                                                    onClick={() =>
-                                                        setIsPhoneVisible(!isPhoneVisible)}
+                                                    onClick={handleNumberVisible}
```
Comment (akarjakin) [inline apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx:185]: More straight forward strategy here would be handling only flag value by this callback, and format value according to this flag, using useMemo

**Comment by gegius** [comment]
Reverted useEffect back as we need to call setShowPhone on start

File: apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx [hunk 3]
```diff
@@ -144,12 +144,13 @@ export const ValidatePhoneForm = ({ onFinish, onReset, title }): React.ReactElem
...
-        } else {
-            const unHidden = formatPhone(phone)
-            setShowPhone(`${unHidden.substring(0, 9)}***${unHidden.substring(12)}`)
-        }
+        const formattedPhone = formatPhone(phone)
+        const phoneVisible =
+            isPhoneVisible ?
+                formattedPhone
+                :
+                `${formattedPhone.substring(0, 9)}***${formattedPhone.substring(12)}`
```
Comment (pahaz) [inline apps/condo/domains/user/components/auth/ValidatePhoneForm.tsx:152]:
```suggestion
        const phoneVisible = isPhoneVisible ? formattedPhone : `${formattedPhone.substring(0, 9)}***${formattedPhone.substring(12)}`
```

**Review by pahaz — APPROVED** [review]

**Review by sitozzz — APPROVED** [review]

**Pull request merged**